xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/ModelBridge.java

```diff
@@ -136,22 +137,14 @@
 
     /**
      * @param deletedDocumentId the ID of the deleted document to restore
-     * @param checkContextUser {@code true} if rights should be checked for the context user, {@code false} otherwise
+     * @param checkRightsRequest the request containing info about rights to check
      * @return {@code true} if the document was restored successfully, {@code false} if the restore failed
      */
-    boolean restoreDeletedDocument(long deletedDocumentId, boolean checkContextUser);
+    boolean restoreDeletedDocument(long deletedDocumentId, AbstractCheckRightsRequest checkRightsRequest);
```

Comment:
Since you change the signature you essentially introduced a new method so a `@since` would be nice.